ci/ray_ci/tester_container.py

```diff
@@ -113,6 +118,39 @@ def _upload_build_info(self, bazel_log_dir) -> None:
             ]
         )
 
+    def _upload_test_results(self, bazel_log_dir: str) -> None:
+        for test, result in self._get_test_and_results(bazel_log_dir):
+            # TODO(can): upload test results
+            logger.info(
+                f"Updating test object {test.get_name()} with result {result.status}"
+            )
+
+    def _get_test_and_results(
+        self, bazel_log_dir: str
+    ) -> List[Tuple[Test, TestResult]]:
+        bazel_logs = []
+        # Find all bazel logs
+        for file in listdir(bazel_log_dir):
+            log = path.join(bazel_log_dir, file)
+            if path.isfile(log) and file.startswith("bazel_log"):
+                bazel_logs.append(log)
+
+        test_and_results = []
+        # Parse bazel logs and print test results
+        for file in bazel_logs:
+            with open(file, "rb") as f:
+                for line in f.readlines():
```

Comment:
you can just `for line in f`, which will read the lines in a streaming fashion.

`readlines()` will read all lines into memory, and it can be huge sometimes.

Reply:
nice